src/kernels.jl

```diff
@@ -94,6 +101,17 @@ function _transduce!(buf, rf::F, init, arrays...) where {F}
     blocks = cld(n, basesize * threads)
     @assert blocks <= kernel_config.blocks
 
+    if Base.issingletontype(acctype)
```

Comment:
for handling https://github.com/JuliaGPU/CUDA.jl/issues/751 (and it's better for performance anyway)